src/main/kotlin/com/sourcegraph/cody/chat/ui/MessagesPanel.kt

```diff
@@ -24,10 +26,14 @@ class MessagesPanel(private val project: Project) :
     removeBlinkingCursor()
 
     if (componentCount > 0) {
-      val lastPanel = components.last() as? JPanel
-      val lastMessage = lastPanel?.getComponent(0) as? SingleMessagePanel
-      if (message.id == lastMessage?.getMessageId()) {
-        lastMessage.updateContentWith(message)
+      val messageToUpdate =
+          components
+              .mapNotNull { (it as? JPanel) }
+              .filter { it.componentCount > 0 }
+              .map { it.getComponent(0) as? SingleMessagePanel }
+              .find { it?.getMessageId() == message.id }
```

Comment:
unrel fix - `find` instead of `last` (so we can update the message wherever it is)